### DISCUSSION TITLE
Create user and send them an email of type magiclink using generateLink

### DISCUSSION THREAD

**Discussion by JUNIORCO** [supabase/supabase#11241] [Questions] [answered]
I want to create an unconfirmed auth user and send them a magic link with a 6-digit OTP (note: it HAS to be an email of type magic link). Then in the app I'll use `supabase.auth.verifyOtp({ email, token: OTP, type: 'signup' })`, which will verify the OTP from the email & mark the email as confirmed.

I'm trying to do this using the admin function `generateLink`. It creates an unconfirmed auth user, but doesn't send out an email and I'm not sure why. Code snippet below.

```
await supabase.auth.admin.generateLink({
  type: 'magiclink',
  email,
  options: {
    data: { ... },
  },
});
```

Does `generateLink` not send an email? If it doesn't, anyone have any tips to achieve the flow I described above?

**Comment by GaryAustin1** [comment, marked as answer]
generateLink just returns you a link you can use in your own email process, I believe...
`Generates email links and OTPs to be sent via a custom email provider.`

There is  inviteUserByEmail that is probably what you want.
Edit:  I guess it uses the invite template though, not magiclink...

I say probably, because I've not actually used it. 

Edit2:
Why can't use just use signInWithOtp?  All it needs is an email and by default it creates a user if they don't exist.
